webrtc.html

```diff
@@ -5331,22 +5330,46 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
                 <code>InvalidStateError</code>.</li>
                 <li>If <code><var>parameters</var>.encodings.length</code>
                 is different from <var>N</var>, or if
-                any parameter in the <var>parameters</var> argument,
+                any parameter in <var>parameters</var>, is
                 marked as a <dfn>Read-only parameter</dfn>, has a value that is
                 different from the corresponding parameter value in <var>sender</var>'s
                 <a>[[\LastReturnedParameters]]</a> internal slot,
-                abort these steps and return a promise rejected with a newly
+                return a promise rejected with a newly
                 <a data-link-for="exception" data-lt="create">created</a>
                 <code>InvalidModificationError</code>. Note that this also
                 applies to <var>transactionId</var>.</li>
                 <li>If the <code>scaleResolutionDownBy</code> parameter in the
-                <var>parameters</var> argument has a value less than 1.0, abort
-                these steps and return a promise rejected with a newly
+                <var>parameters</var> argument has a value less than 1.0,
+                return a promise rejected with a newly
                 <a data-link-for="exception" data-lt="create">created</a>
                 <code>RangeError</code>.</li>
-                <li>Set <var>sender</var>'s internal <a>[[\SendEncodings]]</a>
-                slot to <code><var>parameters</var>.encodings</code>.</li>
-                <li>Resolve <var>p</var> with <code>undefined</code>.</li>
+                <li>Let <var>p</var> be a new promise.</li>
+                <li>In parallel, configure the media stack to use
+                <var>parameters</var> to transmit <var>sender</var>.
+                  <ol>
+                    <li>If the media stack is successfully configured with
+                    <var>parameters</var>, queue a task to run the following
+                    steps:
+                      <ol>
+                        <li>Set <var>sender</var>'s internal
+                        <a>[[\SendEncodings]]</a> slot to <code>
+                        <var>parameters</var>.encodings</code>.</li>
+                        <li>Resolve <var>p</var> with <code>undefined</code>.
+                      </ol>
+                    </li>
+                    <li>If any error occurred while configuring the media stack,
+                    queue a task to run the following steps:
+                      <ol>
+                        <li>If an error occurred due to hardware resources not
+                        being available, reject <var>p</var> with a newly
+                        created <code><a>RTCError</a></code> whose
+                        <code>errorDetail</code> set to
+                        "hardware-encoder-error".</li>
+                      </ol>
```

Comment:
Do we need a catchall "OperationError" if something non-hardware-related failed in the async steps?

Reply:
Let's add that. Hopefully it wouldn't be used much if we have a rather complete list of specific errors.